multimedia/VLC/Portfile

```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
-                    port:libupnp \
-                    port:libusb \
-                    port:libvorbis \
-                    port:libvpx \
-                    port:libxml2 \
-                    port:live555 \
-                    port:ncurses \
-                    port:nettle \
-                    port:opencv \
-                    port:openjpeg15 \
-                    path:lib/libssl.dylib:openssl \
-                    port:orc \
-                    port:p11-kit \
-                    port:readline \
-                    port:schroedinger \
-                    port:speex \
-                    port:taglib \
-                    port:tiff \
-                    port:twolame \
-                    port:vcdimager \
-                    port:x264 \
-                    port:x265
-
-# the +qt5 and/or +qt4 variants of port:opencv currently lead to build errors
-pre-configure {
-    if {![active_variants opencv "" {qt4 qt5}]} {
-        ui_error "${subport} cannot currently be built against port:opencv with the qt4 or qt5 variant.\
-        Install port:opencv without either qt variant and try again. This is only a build conflict; \
-        after building you can reactivate your preferred opencv variant"
-        return -code error "Install or activate port:opencv without a qt variant before building ${subport}"
+# VLC-2.1 and later are x86_64-only: https://trac.videolan.org/vlc/ticket/8161
+universal_variant       no
+supported_archs         x86_64
+
+##
+###################  ports VLC and libVLC  ###################
+##
+if {(${subport} eq ${name}) || (${subport} eq "lib${name}")} {
+    version             2.2.4
+    license             GPL-2
+
+    platforms           darwin
+
+    master_sites        http://download.videolan.org/pub/videolan/vlc/${version}/
+    distname            vlc-${version}
+    use_xz              yes
+
+#     checksums           rmd160  8c6d4194da37210fe409c965f430898fbe7f2416 \
+#                         sha256  b9d7587d35f13c3c981964c8cc8b03f1c7c8edf528be476b3ca1d2efedd5bf5b
+    checksums           rmd160  2fe178bb9a4b2ad5ca0294d3e52ceb992b22e8ee \
+                        sha256  1632e91d2a0087e0ef4c3fb4c95c3c2890f7715a9d1d43ffd46329f428cf53be
+
+    if {[info exist ::env(MACPORTS_COMPRESS_WORKDIR)] && $::env(MACPORTS_COMPRESS_WORKDIR)} {
+        # Enable HFS compression.
+        if {[file exists ${prefix}/bin/bsdtar]} {
+            extract.post_args \
+                        "| ${prefix}/bin/bsdtar -x --hfsCompression"
+        }
     }
-}
 
-# VLC 2.2.2 should have full Lua 5.3 compatibility, for now, depend on port:lua52
-# That requires the patch-for-lua52.diff which is appended to patchfiles below.
-# update 2.2.2 : lua 5.3 gives compiler syntax errors so we stick with 5.2 for now.
-depends_lib-append  port:lua52
-# depends_lib-append  port:lua
+    depends_build       port:pkgconfig
+
+    depends_lib         path:${FFMPEG_VLC_PREFIX}/lib/libavcodec.dylib:ffmpeg-VLC \
```

Comment:
Isn't the `ffmpeg-VLC` port the only port that provides this file anyway? If so, why have the `path:`-style dependency?